### DISCUSSION TITLE
Proxy from Source HTTPS to target HTTP

### DISCUSSION THREAD

**Discussion by siddiqicps** [chimurai/http-proxy-middleware#544] [Q&A]
Dear Team,

I am running a node.js webserver over https and my node.js APIs are running over HTTP. I am trying to proxy from source HTTPS to target HTTP but it's not working. Is there any specific option which I have to set in order to achieve this.

**Comment by chimurai** [comment]
try setting the `ssl` option:

```javascript
  ssl: {
    key: fs.readFileSync(path.join(fixturesDir, 'agent2-key.pem'), 'utf8'),
    cert: fs.readFileSync(path.join(fixturesDir, 'agent2-cert.pem'), 'utf8')
  }
```

source:
https://github.com/http-party/node-http-proxy/blob/master/examples/http/proxy-https-to-http.js#L53

**Comment by siddiqicps** [comment]
Thanks, but stringifying the request payload solved my problem.

```
`function onProxyReq(proxyReq, req, res) {
   if (req.body && req.complete) {
      **const bodyData = JSON.stringify(req.body);**
      proxyReq.write(bodyData);
    }
 }`
```
